roofit/roofitcore/test/testTestStatistics.cxx

```diff
@@ -477,6 +481,67 @@ TEST_P(OffsetBinTest, CrossCheck)
    EXPECT_NEAR(nllVal1, 0.0, 1e-8) << "NLL with bin offsetting is " << nllVal1 << ", and " << nllVal0 << " without it.";
 }
 
+// Verify that the binned likelihood optimization works also when fitting a
+// single-channel RooRealSumPdf or RooProdPdf.
+TEST_P(TestStatisticTest, BinnedLikelihood)
+{
+   using namespace RooFit;
+
+   int nEvents = 1000;
+   int numBins = 5;
+
+   RooWorkspace ws;
+
+   ws.factory("x[0, 0, " + std::to_string(numBins) + "]");
+
+   auto &x = *ws.var("x");
+   x.setBins(numBins);
+
+   {
+      // Uniform RooDataHist
+      RooDataHist dataHist{"data_hist", "data_hist", x};
+      for (int iBin = 0; iBin < numBins; ++iBin) {
+         dataHist.set(iBin, nEvents / numBins, -1);
+      }
+
+      RooHistFunc histFunc{"hist_func", "hist_func", x, dataHist};
+      RooRealSumPdf pdf{"pdf", "pdf", histFunc, RooArgList{1.0}};
+
+      // Enable the binned likelihood optimization to avoid integrals
+      // (like in HistFactory).
+      pdf.setAttribute("BinnedLikelihood");
+
+      // Wrap the channel pdf in a RooProdPdf to mimic HistFactory
+      RooProdPdf prodPdf{"prod_pdf", "prod_pdf", pdf};
+
+      ws.import(prodPdf);
+   }
+
+   // We have to wrap the pdf in a RooSimultaneous, otherwise the
+   // BinnedLikelihood optimization doesn't work.
```

Comment:
isn;'t this comment misleading at this point?

Reply:
You're right, thanks for catching this